torch/fx/passes/utils/matcher_utils.py

```diff
@@ -350,3 +350,52 @@ def backtracking(anchor_index, match):
         logger.info(f"Matches returned: {matches}")
 
         return matches
+
+
+@compatibility(is_backward_compatible=False)
+def replace_literals_with_placeholders(graph: torch.fx.Graph) -> torch.fx.Graph:
+    """
+    Replaces all the literals that are passed into call_function nodes in the
+    graph as placeholders in the graph. This way during matching these literals
```

Comment:
This feels like an hack to work around the problem. :) 

I wonder how this util will be used together with SubgraphMatcher? Is this pass expected to be applied on the target graph, or pattern graph? This pass is intrusive, and make the graph un-runableable after the replacement. So I guess this is expected to be applied on the pattern graph? 

I think the real problem is to "allow treating literals as wildcard" in SubgraphMatcher. and the real fix should be also inside SubgraphMatcher.

Reply:
Yeah, this is to be used on a pattern graph. Context is that when tracing torch.nn.Linear it'll call some functions with different inputs depending on size of the input tensors. We don't want to generate a different pattern each time we pass in a different sized input, so instead will convert the pattern graph to be more generic by replacing literals with the placeholders. 

> allowing treating literals as wildcards in SubgraphMatcher

I'm fine with adding a flag to SubgraphMatcher to skip over literals if you think this is better.

Reply:
I think it's better to add a flag, assuming it doesn't add too much complexity to our SubgraphMatcher code.

Reply:
> I think it's better to add a flag, assuming it doesn't add too much complexity to our SubgraphMatcher code.

Yeah this would be great.